internal/hud/server/websocket_reader.go

```diff
@@ -0,0 +1,174 @@
+package server
+
+import "net/url"
+
+// A thinger to connect to the HUD server websocket, for CLI commands that
+// want to read state from a running Tilt (e.g. `tilt logs`).
+// TODO(maia): figure out if this should live here, or elsewhere (I assume
```

Comment:
My rule of thumb when it comes to such things is to wait for the second time I need it. We're more likely to have a better idea of what it should look like then, and I wouldn't want to hold things up in the meantime.

Reply:
fair nough -- it would be MORE work now to re-merge the types lol, but I changed the note up top to reflect this!